library/src/main/java/com/chuckerteam/chucker/api/ChuckerInterceptor.kt

```diff
@@ -235,30 +235,27 @@ class ChuckerInterceptor internal constructor(
         private val response: Response,
         private val transaction: HttpTransaction
     ) : TeeSource.Callback {
-        override fun onSuccess(file: File) {
-            val buffer = readResponseBuffer(file, response.isGzipped)
-            file.delete()
+        override fun onClosed(file: File) {
+            val buffer = try {
+                readResponseBuffer(file, response.isGzipped)
+            } catch (e: IOException) {
+                null
+            }
             if (buffer != null) processResponseBody(response, buffer, transaction)
             collector.onResponseReceived(transaction)
-        }
-
-        override fun onFailure(exception: IOException, file: File) {
             file.delete()
-            collector.onResponseReceived(transaction)
         }
 
-        private fun readResponseBuffer(responseBody: File, isGzipped: Boolean): Buffer? {
+        override fun onFailure(exception: IOException, file: File) = Unit
```

Comment:
Should you delete the file here?

Reply:
Not anymore. Now, `onClosed` is called whenever upstream is closed. Even when there was a failure in the meantime. Deleting file here would make it unavailable in the `onClosed` callback.

Reply:
Unless you want to `onClosed` be called only when failure did not occur. I changed the name from `onSuccess` to `onClosed` because now it is hard to know what success really is since we no longer rely on the upstream being exhausted.

Reply:
Yeah I think now with `onClosed` is clearer 👍